### PR TITLE
Gets CMake working on Windows

### PR DESCRIPTION
**Pull request by lokitoth**
* Also enables building dotnet bindings via CMake on Windows

### CODE REVIEW

File: cmake/platforms/win32.cmake [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+if(WIN32)
+  # VW targets Windows 10.0.16299.0 SDK
+  set(CMAKE_SYSTEM_VERSION "10.0.16299.0" CACHE INTERNAL "Windows SDK version to target.")
+
+  # TODO: Enable building .NET bindings via CMake on Linux 
+  option(rlclientlib_BUILD_DOTNET "Build .NET bindings" ON)
+
+  if (rlclientlib_BUILD_DOTNET)
+    cmake_minimum_required(VERSION 3.14)
+    # set(CMAKE_DOTNET_TARGET_FRAMEWORK_VERSION  "v4.5.2" CACHE INTERNAL ".NET Framework SDK version to target.")
```
Comment (olgavrou) [inline cmake/platforms/win32.cmake:20]: remove commented out code

File: CMakeLists.txt [hunk 2]
```diff
@@ -14,7 +22,7 @@ if (vw_USE_AZURE_FACTORIES)
 endif()
 
 find_package(OpenSSL REQUIRED)
-find_package(Boost COMPONENTS unit_test_framework system program_options thread REQUIRED)
+find_package(Boost COMPONENTS unit_test_framework system program_options thread REQUIRED) # Boost::uuid too, but that cannot be "found" here
```
Comment (olgavrou) [inline CMakeLists.txt:25]: where is it found?

File: bindings/cs/CMakeLists.txt [hunk 3]
```diff
@@ -1 +1,29 @@
-add_subdirectory(rl.net.native)
+if (rlclientlib_BUILD_DOTNET)
+  set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/Modules/")
+  include(FindDotnet)
+
+  add_subdirectory(rl.net.native)
+
+  add_custom_target(rl.net
```
Comment (jackgerrits) [inline bindings/cs/CMakeLists.txt:7]: Can you add the sources to these targets so when a VS solution is generated they show up?

File: CMakeLists.txt (hunk 2 above)
Comment (lokitoth) [reply CMakeLists.txt:25]: It is a header-only library, which cannot be found as a component by find_package()

File: bindings/cs/CMakeLists.txt (hunk 3 above)
Comment (lokitoth) [reply bindings/cs/CMakeLists.txt]: Done. At that point it makes sense to break them out into separate CMakeLists.txt files. 

It is somewhat of an impedance mismatch between CMake and the dotnet toolchain, as .Net developers will not naturally think to add new sources into the CMakeLists.txt file. That was one of the big gains in Sdk-style projects: including all source in the tree by default.

**Review by jackgerrits — APPROVED** [review]

**Review by olgavrou — APPROVED** [review]
LGTM :)

**Pull request merged**